playlist/playlist.go

```diff
@@ -204,3 +204,33 @@ func (p *Playlist) FindNodeInsideAdBreak(node *internal.Node) (*internal.Node, b
 
 	return nil, false
 }
+
+// Returns the last DateRange (#EXT-X-DATERANGE) node with SCTE35-OUT marking (i.e. the last Ad Break) in the playlist
+func (p *Playlist) FindLastAdBreak() (*internal.Node, bool) {
+	adBreaks := p.Breaks()
+	if len(adBreaks) == 0 {
+		return nil, false
+	}
+	return adBreaks[len(adBreaks)-1], true
+}
+
+// DuplicateAdBreak checks if two ad breaks have the same START-DATE, indicating a duplicate.
+// Two ad breaks are considered duplicates if they share the same START-DATE
+// and the same PLANNED-DURATION.
+func (p *Playlist) FindDuplicateAdBreak() bool {
```

Comment:
Instead of the prefix `Find`, you could use `Has`, because `Find` implies you are retrieving an object, but your method is merely a boolean check (if the duplicate ad break exists or not).

```suggestion
func (p *Playlist) HasDuplicateAdBreak() bool {
```